source/agora/consensus/data/Block.d

```diff
@@ -63,6 +65,9 @@ public struct BlockHeader
         dg(this.prev_block[]);
         hashPart(this.height, dg);
         dg(this.merkle_root[]);
+
+        foreach (validator_sig; this.validator_sigs)
```

Comment:
The validators sign the block header hash, correct? But the block header hash includes the signatures. So how do you sign a block header hash when that hash includes the signature? You can't, because as soon as you sign a block header and then set the signature, the block header hash changes.

You cannot include the validator signatures as part of the block header hash.

Reply:
ValidatorSig -
`    public void computeHash (scope HashDg dg) const nothrow @safe @nogc
    {
        hashPart(this.key, dg);
        dg(this.utxo_hash[]);
    }`
Signature information is excluded from computeHash.